src/reflex.cpp

```diff
@@ -1526,6 +1549,10 @@ void Reflex::write()
     options["YYLTYPE"] = options["bison_cc_namespace"] + "::location";
   if (!options["bison_cc"].empty() && options["YYSTYPE"].empty())
     options["YYSTYPE"] = options["bison_cc_namespace"] + "::" + options["bison_cc_parser"] + "::semantic_type";
+  if (!options["bison_complete"].empty() && options["token_type"].empty())
+    options["token_type"] = options["bison_cc_namespace"] + "::" + options["bison_cc_parser"] + "::symbol_type";
+  if (options["token_type"].empty())
```

Comment:
I suggest to replace lines 1554+1555 by:
~~~
std::string token_type = options["token_type"].empty() ? "int" : options["token_type"];
~~~
This changes aligns with `yyltype` and `yystype` at line 1529. Then remove line 1873 and rename `ret_type` by `token_type`. This change generates a `REFLEX_OPTION_token_type` macro in lex.yy.cpp only if the option is actually specified and does not otherwise, which is cleaner.